docs/devguide/contributing.rst

```diff
@@ -35,3 +35,86 @@ negative votes, the team will grant commit privileges.
 The core team will also be responsible for removing commit privileges when
 appropriate - for example for malicious merge behavior or just inactivity over
 an extended period of time.
+
+
+Quality gates for the pull requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are three quality gates to ensure the pull requests quality, `Hound`_ for
+code style check, `Travis CI`_ for unit-test and coveralls, `Jenkins`_ for the combination 
+test including unit-test and smoke test.. When a pull request created, all tests 
+will run automatically, test results can be found in the merge status field of
+each pull request page.
+
+.. _Hound: https://houndci.com/
+.. _Travis CI: https://travis-ci.org/
+.. _Jenkins: https://jenkins.io/
+.. _jshint: http://jshint.com/
+
+**Hound**
+
+Hound works with `jshint`_ and comments on style violations in pull requests. 
+Configuration files ``.hound.yml`` and ``.jshintrc`` have been created in each 
+repository, so before creating a pull request, you can check code style locally with 
+jshint to find out style violations beforehand.
+
```

Comment:
What about adding some additional wording within this new section ("Quality gates for the pull requests") that strongly encourages folks to run unit / functional tests locally prior to creating a PR.  This would hopefully minimize the amount errors seen during PR submission and lessen a dependency on Travis / Jenkins to test code before it's really ready to be submitted.